flash.sh

```diff
@@ -0,0 +1,285 @@
+ADB=${ADB:-adb}
+VARIANT=${VARIANT:-eng}
+FULLFLASH=false
+SLICE="p"
+BS_SLICE="M"
+BMAPTOOL=false
+
+if [ ! -f "`which \"$ADB\"`" ]; then
+	ADB=out/host/`uname -s | tr "[[:upper:]]" "[[:lower:]]"`-x86/bin/adb
+fi
+
+run_adb()
+{
+	$ADB $ADB_FLAGS $@
+}
+
+get_drive_slice()
+{
+	case `uname` in
+  "Darwin")
+    SLICE="s"
+    ;;
+  "Linux")
+    SLICE="p"
+    ;;
+  *)
+    echo Unsupported platform: `uname`
+    exit -1
+  esac
+}
+
+get_drive_slice
+
+get_bs_slice()
+{
+	case `uname` in
+  "Darwin")
+    BS_SLICE="m"
+    ;;
+  "Linux")
+    BS_SLICE="M"
+    ;;
+  *)
+    echo Unsupported platform: `uname`
+    exit -1
+  esac
+}
+
+get_bs_slice
+
+is_bmaptool()
+{
+	bmaptool
+	if [ $? -eq 0 ]; then
+	    BMAPTOOL=true
+	else
+	    BMAPTOOL=false
+	fi
+}
+
+is_bmaptool
+
+update_time()
+{
+	if [ `uname` = Darwin ]; then
+		OFFSET=`date +%z`
+		OFFSET=${OFFSET:0:3}
+		TIMEZONE=`date +%Z$OFFSET|tr +- -+`
+	else
+		TIMEZONE=`date +%Z%:::z|tr +- -+`
+	fi
+	echo Attempting to set the time on the device
+	run_adb wait-for-device &&
+	run_adb shell "toolbox date $(date +%s) ||
+	               toolbox date -s $(date +%Y%m%d.%H%M%S)" &&
+	run_adb shell setprop persist.sys.timezone $TIMEZONE
+}
+
+sd_partition()
+{
+  DRIVE=$2
+
+  if [ -z "${DRIVE}" ]; then
+    echo You must specify a SD card drive
+  fi
+
+  umount $DRIVE
+
+  dd if=/dev/zero of=$DRIVE bs=1024
+
+  sed -e 's/\t\([\+0-9a-zA-Z]*\)[ \t].*/\1/' << EOF | fdisk ${DRIVE}
+    o # clear the in memory partition table
+    n # new partition
+    p # primary partition
+    1 # partition number 1
+      # default - start at beginning of disk
+    +48M # 48 MB boot parttion
+    n # new partition
+    p # primary partition
+    2 # partition number 2
+      # default, start immediately after preceding partition
+    +537M # 537 MB boot parttion
+    n # new partition
+    p # primary partition
+    3 # partition number 3
+      # default, start immediately after preceding partition
+    +128M # 128 MB boot parttion
+    n # new partition
+    p # primary partition
+    4 # partition number 4
+      # default, start immediately after preceding partition
+      # default, extend partition to end of disk
+    a # make a partition bootable
+    1 # bootable partition is partition 1 -- /dev/xxxx1
+    p # print the in-memory partition table
+    w # write the partition table
+    q # finish
+  EOF
+
+	e2fsck -f ${DRIVE}${SLICE}4
+
+  resize2fs -f ${DRIVE}${SLICE}4
+
+}
+
+flash_image()
```

Comment:
This flash_image() does too much at a time. Please split it in several functions:
 - one to flash one partition,
 - one to check sd